dd-sdk-android-compose/src/main/kotlin/com/datadog/android/compose/InteractionTracking.kt

```diff
@@ -55,3 +158,273 @@ internal class TapActionTracker(
+) {
+    trackDragInteraction<ScrollStartProps>(
+        interactionSource,
+        onStart = { interactions, start ->
+            interactions[start] =
+                ScrollStartProps(
+                    interactionType.scrollableState.currentPosition
+                )
+            rumMonitor.startUserAction(RumActionType.SCROLL, targetName, emptyMap())
+        },
+        onStopOrCancel = { startProps ->
+            reportScrollInteraction(
+                rumMonitor,
+                targetName,
+                startProps,
+                interactionType,
+                isRtl,
+                attributes
+            )
+        }
+    )
+}
+
+internal const val FROM_SWIPE_STATE_ATTRIBUTE = "from_state"
+internal const val TO_SWIPE_STATE_ATTRIBUTE = "to_state"
+
+// endregion
+
+// region private
+
+private suspend fun <T> trackDragInteraction(
+    interactionSource: InteractionSource,
+    onStart: (
+        interactions: MutableMap<DragInteraction.Start, T>,
+        start: DragInteraction.Start
+    ) -> Unit,
+    onStopOrCancel: (startProps: T) -> Unit
+) {
+    val ongoingInteractions = mutableMapOf<DragInteraction.Start, T>()
+    try {
+        interactionSource.interactions.collect { interaction ->
+            when (interaction) {
+                is DragInteraction.Start -> {
+                    @Suppress("UnsafeThirdPartyFunctionCall")
+                    onStart(ongoingInteractions, interaction)
+                }
+                is DragInteraction.Stop, is DragInteraction.Cancel -> {
+                    val start = when (interaction) {
+                        is DragInteraction.Stop -> interaction.start
+                        is DragInteraction.Cancel -> interaction.start
+                        else -> {
+                            // this will never happen, let's just make compiler happy and
+                            // give a fake start
+                            Log.e(
+                                LOG_TAG,
+                                "Unexpected branch reached" +
+                                    " for the drag interaction start"
+                            )
+                            DragInteraction.Start()
+                        }
+                    }
+
+                    ongoingInteractions.remove(start)?.let {
+                        @Suppress("UnsafeThirdPartyFunctionCall")
+                        onStopOrCancel(it)
+                    }
+                }
+            }
+        }
+    } catch (@Suppress("TooGenericExceptionCaught") e: Exception) {
+        Log.e(LOG_TAG, "Exception during drag interactions tracking", e)
+    }
+}
+
+private val ScrollableState.currentPosition: Int?
+    get() {
+        return when (this) {
+            is LazyListState -> {
+                this.firstVisibleItemIndex
+            }
+            is ScrollState -> {
+                this.value
+            }
+            else -> {
+                null
+            }
```

Comment:
they are different indeed (index vs offset in pixels), but since they follow the same logic - when scrolling down both index and offset decrease, up - increase, they can be unified behind the common property `position`. `ScrollableState` interface itself doesn't expose any field which can tell about position unfortunately.